core/src/main/scala/org/apache/spark/status/AppStatusListener.scala

```diff
@@ -917,8 +917,24 @@ private[spark] class AppStatusListener(
     // Update the block entry in the RDD info, keeping track of the deltas above so that we
     // can update the executor information too.
     liveRDDs.get(block.rddId).foreach { rdd =>
+
       if (updatedStorageLevel.isDefined) {
-        rdd.setStorageLevel(updatedStorageLevel.get)
+        // Replicated block update events will have `storageLevel.replication=1`.
```

Comment:
Is this a bug itself?

Reply:
Needs more check, including impacts. Currently the fix is from UI side.